lib/datasource/artifactory/index.ts

```diff
@@ -0,0 +1,111 @@
+import { logger } from '../../logger';
+import { cache } from '../../util/cache/package/decorator';
+import { parse } from '../../util/html';
+import { HttpError } from '../../util/http/types';
+import { Datasource } from '../datasource';
+import type { GetReleasesConfig, Release, ReleaseResult } from '../types';
+import { datasource } from './common';
+
+export class ArtifactoryDatasource extends Datasource {
+  static readonly id = datasource;
+
+  constructor() {
+    super(datasource);
+  }
+
+  override readonly customRegistrySupport = true;
+
+  override readonly caching = true;
+
+  override readonly registryStrategy = 'merge';
+
+  @cache({
+    namespace: `datasource-${datasource}`,
+    key: ({ registryUrl }: GetReleasesConfig) => `${registryUrl}`,
```

Comment:
```suggestion
    key: ({ registryUrl, lookupName }: GetReleasesConfig) => `${registryUrl}:${lookupName}`,
```

`lookupName` needs to be added to the cache key, otherwise we've overlappings. 😉